src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransferByte.java

```diff
@@ -60,7 +60,11 @@ public SerializationMode getSerializationMode(final BytesRef bytesRef) {
 
     private void transfer() {
         int lengthOfVector = dimension / 8;
-        vectorAddress = JNICommons.storeByteVectorData(vectorAddress, vectorList.toArray(new byte[][] {}), totalLiveDocs * lengthOfVector);
+        if(totalLiveDocs > 0) {
```

Comment:
This is so that we have an option to only allocate enough memory for one transfer. This is as opposed to allocating enough memory for all of the live docs. This seemed like the easiest way to do this.

Reply:
Resolved as this is still necessary.